docs/getting-started/docker-playgrounds/flink_cluster_playground.md

```diff
@@ -25,3 +25,183 @@ under the License.
 
 * This will be replaced by the TOC
 {:toc}
+
+
+# Description
+
+* Flink Session Cluster with one JobManager and one TaskManager
+* Flink Client Container
+* Checkpointing to a shared Docker volume
+* Savepoints to a shared Docker volume
+* Two Flink Jobs are submitted on startup:
+    * StateMachineExample
+    * KafkaEventsGeneratorJob
+
+# Deployment
+
+## Prerequisits
+
+* docker
+* docker-compose
+
+## Setup
+
+* docker-compose.yaml and Flink configuration directory needed.
+
+**Directory Structure**
+{% highlight bash %}
+$ tree playground/
+playground/
+├── docker-compose.yaml
+└── conf
+    └── flink-conf.yaml
+    └── log4j-cli.properties
+    └── log4j-console.properties
+{% endhighlight %}
+
+**docker-compose.yaml**
+{% highlight yml %}
+version: "2.1"
+services:
+  client:
+    image: flink:1.8.0-scala_2.11
+    command: '/bin/bash -c "flink run -d /opt/flink/examples/streaming/StateMachineExample.jar --error-rate 0.01 --sleep 10 --backend file --checkpoint-dir file:///tmp/flink-checkpoint-directory --kafka-topic input --brokers kafka:9092
+              && flink run -c org.apache.flink.streaming.examples.statemachine.KafkaEventsGeneratorJob -d /opt/flink/examples/streaming/StateMachineExample.jar --kafka-topic input --brokers kafka:9092"'
+    depends_on:
+      - jobmanager
+      - kafka
+    volumes:
+      - ./conf:/opt/flink/conf
+    environment:
+      - JOB_MANAGER_RPC_ADDRESS=jobmanager
+  jobmanager:
+    image: flink:1.8.0-scala_2.11
+    command: "jobmanager.sh start-foreground"
+    ports:
+      - 8081:8081
+    volumes:
+      - ./conf:/opt/flink/conf
+      - flink-checkpoint-directory:/tmp/flink-checkpoint-directory
+      - /tmp/flink-playground/savepoints-directory:/tmp/flink-savepoints-directory
+    environment:
+      - JOB_MANAGER_RPC_ADDRESS=jobmanager
+  taskmanager:
+    image: flink:1.8.0-scala_2.11
+    depends_on:
+      - jobmanager
+    command: "taskmanager.sh start-foreground"
+    user: root
+    volumes:
+      - ./conf:/opt/flink/conf
+      - flink-checkpoint-directory:/tmp/flink-checkpoint-directory
+      - /tmp/flink-playground/savepoints-directory:/tmp/flink-savepoints-directory
+    environment:
+      - JOB_MANAGER_RPC_ADDRESS=jobmanager
+  zookeeper:
+    image: wurstmeister/zookeeper:3.4.6
+  kafka:
+    image: wurstmeister/kafka:2.11-1.1.1
+    environment:
+      KAFKA_ADVERTISED_LISTENERS: INSIDE://:9092,OUTSIDE://:9094
+      KAFKA_LISTENERS: INSIDE://:9092,OUTSIDE://:9094
+      KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: INSIDE:PLAINTEXT,OUTSIDE:PLAINTEXT
+      KAFKA_INTER_BROKER_LISTENER_NAME: INSIDE
+      KAFKA_CREATE_TOPICS: "input:2:1"
+      KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
+volumes:
+  flink-checkpoint-directory:
+{% endhighlight %}
+
+**log4j-cli.properties**
```

Comment:
When writing we'll probably want a sentence describing why two logger properties. It might not be obvious to new users from the file names.

Reply:
Yes, btw: these are the default log4j properties of Flink, but since we mount the conf directory we override the ones that came with the image.